[Tests] Add RPC budget regression tests

Fuzzbawls opened the pull request:
Adds new regression testing for the following RPC budget commands:
- `preparebudget`
- `submitbudget`
- `getbudgetinfo`

CaveSpectre11 commented on test/functional/rpc_budget.py:
```diff
@@ -0,0 +1,87 @@
+        url = 'test.com'
+        longurl = 'test.com/longer-than-allowed-url-string-should-fail-if-url-is-more-than-64-characters'
+        name = 'testinvalid'
+        longname = 'ThisProposalNameHasMoreThanTwentyCharacters'
```
Test parameters should be exactly on the borders of the testing size.  e.g.
```
url = <57 character URL>
longurl = <58 character URL>
name = <20 character name>
longname = <21 character name>
```
Rather than having plenty of wiggle room in the test case; as a change from `<` to `<=` wouldn't be caught.  (And ensuring the max limit actually will pass the code)

Fuzzbawls replied:
done

CaveSpectre11 commented on test/functional/rpc_budget.py:
```diff
@@ -0,0 +1,87 @@
+                                longname, scheme + url, numcycles, nextsuperblock, address, cycleamount)
+
+        self.log.info("Test with long URL")
+        assert_raises_rpc_error(-8, "Invalid URL:", self.nodes[0].preparebudget,
```
I'm thinking it would be best to test the exact expected result; especially now that I notice the preamble is different between the length error and the scheme error (and thus may end up changed in the future to have them consistent).
`Invalid URL: 92 exceeds limit of 64 characters."` (as the test is written)

As it stands, if the scheme test code changes to "Invalid URL:  check scheme"; the length test will have a false positive.

This also helps point out that the test needs to change in parallel with any changes to the code.

Fuzzbawls replied:
fixed

Fuzzbawls commented:
Rebased on top of master and addressed feedback to test the edges.

Also added earlier validation of the requested payout amount to ensure it is within the allowed range.

CaveSpectre11 commented:
looks good; utACK https://github.com/PIVX-Project/PIVX/commit/164de044054dfcf0096092c2e6d8f708e27fce01
